string target
string project
uint16 version
bool full